### DISCUSSION TITLE
Receiving duplicate products

### DISCUSSION THREAD

**Discussion by studiodrimmel** [swellstores/community#506] [Help] [answered]
Hi everybody,

I'm trying to implement some pagination in my productlist, but I receiving duplicate entries in my list.

My code:
```javascript
return swellNodeClient.get('/products', {
        expand: ['variants', 'categories'],
        ...!!category && { categories: [category] },
        limit: PRODUCTS_GRID_LIMIT,
        page,
        ...!!sort && { sort }
    });
```

This is the actual data being consumed by the swell-node client.
```console
storefront:dev: [0] {
storefront:dev: [0]   expand: [ 'variants', 'categories' ],
storefront:dev: [0]   categories: [ 'sieraden' ],
storefront:dev: [0]   limit: 4,
storefront:dev: [0]   page: 1
storefront:dev: [0] }
storefront:dev: [0] {
storefront:dev: [0]   expand: [ 'variants', 'categories' ],
storefront:dev: [0]   categories: [ 'sieraden' ],
storefront:dev: [0]   limit: 4,
storefront:dev: [0]   page: 2
storefront:dev: [0] }
```

These are the results, 2 are duplicates in this example: Amour & Claudia
<img width="528" alt="Screenshot 2024-02-20 at 23 11 11" src="https://github.com/swellstores/community/assets/101345863/5e5133a7-022a-40b0-a989-c56889423be0">

I can reproduce it in the Console using the following GET requests:
/products?category=Sieraden&limit=4&fields=name&page=1
/products?category=Sieraden&limit=4&fields=name&page=2

**Comment by logeshswell** [comment, marked as answer]
I can confirm that it's a bug. We'll take a deeper look at this issue and get back to you if I get some updates.

**Reply by studiodrimmel** [reply]
Any update on this matter? It is really blocking that we can't filter on a category atm, it doesn't work with either category or categories query parameter so..
?page=1&limit=4&fields=name&categories=sieraden
?page=1&limit=4&fields=name&category=sieraden

**Reply by logeshswell** [reply]
I don't have any ETA at the moment, but I will keep you posted if I get some updates.

**Reply by studiodrimmel** [reply]
Still no update? This is a key part of many ecommerce websites.. How can something like this be broken for so long..

**Reply by logeshswell** [reply]
We identified that the issue arises only when the same product belongs to both parent and child categories.
I don't have any other updates at the moment.

**Comment by logeshswell** [comment]
This issue is fixed and deployed at our end. Can you check and confirm please?

**Reply by studiodrimmel** [reply]
Seems to be working now, had to do a little rewrite to test it. Thanks
